stdlib/LinearAlgebra/test/qr.jl

```diff
@@ -504,4 +504,28 @@ end
     @test x ≈ xf
 end
 
+@testset "issue #53451" begin
+    # in the issue it was noted that QR factorizations of zero-column matrices
+    # were possible, but zero row-matrices errored, because LAPACK does not
+    # accept these empty matrices. now, the `geqrt!` call should be forwarded only
+    # if both matrix dimensions are positive.
+
+    for dimA in (0, 1, 2, 4)
+        for F in (Float32, Float64, ComplexF32, ComplexF64)
```

Comment:
I believe we also have generic `qr` methods, in which case which you could test for this edge case behavior with `BigFloat`, for instance?

Reply:
Aha, yes, at least `qr` works for `F = BigFloat`:

```suggestion
        for F in (Float32, Float64, ComplexF32, ComplexF64, BigFloat)
```